bicep/main.bicep

```diff
@@ -1302,4 +1302,19 @@ output LogAnalyticsName string = (createLaw) ? aks_law.name : ''
 output LogAnalyticsGuid string = (createLaw) ? aks_law.properties.customerId : ''
 output LogAnalyticsId string = (createLaw) ? aks_law.id : ''
 
+//---------------------------------------------------------------------------------- AKS events with Event Grid
+// Supported events : https://docs.microsoft.com/en-gb/azure/event-grid/event-schema-aks?tabs=event-grid-event-schema#available-event-types
+
+@description('Create an Event Grid System Topic for AKS events')
+param createEventGrid bool = false
+
+resource eventGrid 'Microsoft.EventGrid/systemTopics@2021-12-01' = if(createEventGrid) {
+  name: 'evgt-${aks.name}'
+  location: location
+  properties: {
```

Comment:
Sounds great. Maybe later I can work on the sample as well if it isn't assigned to someone already.

Reply:
@ruvyas - Sure thing. Have a look at [this sample](https://github.com/Azure/AKS-Construction/blob/main/samples/shared-acr/main.bicep) for when we have a separate ACR.

I think it'd be similar, it would just create the 
- storage queue / eventhub
- roleassignment for the EventGrid identity on the new resource

I have [asked the AKS team](https://github.com/Azure/AKS/issues/1812#issuecomment-1134542269) about the testing of this feature too.

Reply:
@Gordonby Is there an update on how we can test the Event Grid for `NewKubernetesVersionAvailable` notification? 

I have the sample ready and deployed for 6 hours, haven't received any events yet.